### PR TITLE
Prepare for AT 16.0-0 Beta1

### PR DESCRIPTION
**Pull request by er-1**

### CODE REVIEW

**Comment by er-1** [comment]
AT next, it failed while building python (P10 optimized):
```
In function ‘_bufferedreader_read_generic’,
    inlined from ‘_io__Buffered_read_impl’ at /home/tcbot/bot-worker/at/src/workdir/sources/python/Modules/_io/bufferedio.c:885:0,
    inlined from ‘_io__Buffered_read’ at /home/tcbot/bot-worker/at/src/workdir/sources/python/Modules/_io/clinic/bufferedio.c.h:169:0:
/home/tcbot/bot-worker/at/src/workdir/sources/python/Modules/_io/bufferedio.c:1695: warning: ‘memcpy’ specified bound between 9223372036854775808 and 18446744073709551613 exceeds maximum object size 9223372036854775807 [-Wstringop-overflow=]
 1695 |             memcpy(out + written, self->buffer + self->pos, r);
      | 
./Programs/_freeze_module importlib._bootstrap /home/tcbot/bot-worker/at/src/workdir/sources/python/Lib/importlib/_bootstrap.py Python/frozen_modules/importlib._bootstrap.h
make[1]: *** [Makefile:1201: Python/frozen_modules/importlib._bootstrap.h] Illegal instruction
```

**Comment by mscastanho** [comment]
More details on the Python issue: Python 3.11 added a helper program (`_freeze_module`) to create frozen modules during the build process. The Python Makefile is not detecting `python_power10` as a cross-compilation (probably because host and target archs are the same), so it builds this file with the same flags used for the rest of the build, which in this case includes `-mcpu=power10`:
```
/home/mscastanho/AT/at16-beta1/opt/at-next-17.0-0-alpha/bin/gcc -m64 -c  -DNDEBUG -O3 -mcpu=power10 <lots-of-other-flags> 
/home/mscastanho/AT/at16-beta1/build/at17.0-0-alpha.debian-11_ppc64le_ppc64le/sources/python/Programs/_freeze_module.c
```
Naturally, when we're not building on a Power10, `_freeze_module` binary fails with `Illegal Instruction` when executed.

When cross-compiling it uses a pure-Python version of this same program, which is executed by a "build Python" interpreter (can be specified with `--with-build-python`). But again, this is not an option since the Makefile is not detecting this as a cross-compilation. This will likely require more investigation and some changes to how Python is built for AT next, so it's better suited for a separate PR.

File: configs/16.0/packages/libhugetlbfs/sources [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 #
-# Copyright 2022 IBM Corporation
+# Copyright 2021 IBM Corporation
```
Comment (mscastanho) [inline configs/16.0/packages/libhugetlbfs/sources:3]: Looks like a spurious change.

**Review by mscastanho — CHANGES_REQUESTED** [review]
Found a minor typo in the commit message of `Update gdb on AT 16.0` : `gdb-12-brnahc`. See other comments below.

File: configs/16.0/packages/libhugetlbfs/sources (hunk 1 above)
Comment (er-1) [reply configs/16.0/packages/libhugetlbfs/sources:3]: Fixed

File: configs/16.0/packages/zlib/sources [hunk 2]
```diff
@@ -40,8 +39,6 @@ ATSRC_PACKAGE_TARS=""
 ATSRC_PACKAGE_MAKE_CHECK=""
 ATSRC_PACKAGE_DISTRIB=no
 ATSRC_PACKAGE_BUNDLE=toolchain_extra
-# Not needed because file follows master branch
-# ATSRC_PACKAGE_UPSTREAM=
 
 atsrc_get_patches ()
 {
```
[the remark this answers is not in the document]
Comment (er-1) [reply configs/16.0/packages/zlib/sources:44]: If it's ok and if the current builds pass, I'd like to add those patches in another PR just for zlib.

[the remark this answers is not in the document]
Comment (mscastanho) [reply configs/16.0/packages/zlib/sources:44]: No problem for me. This PR is already big indeed.

**Pull request merged**